web/Dockerfile

```diff
@@ -6,11 +6,12 @@ RUN mkdir -p $APP_DIR/bin
 RUN mkdir -p $APP_DIR/src
 WORKDIR $APP_DIR
 
-RUN npm install -g @angular/cli@6.1.3 --silent --depth 1
-COPY package.json $APP_DIR
-RUN npm install --silent --depth 0
+# this is the NPM cache
+VOLUME /root/.npm
 
-COPY angular.json karma.conf.js protractor.conf.js tsconfig.json tslint.json $APP_DIR
+RUN npm install -g @angular/cli@6.1.3 --depth 1
+
+COPY package.json angular.json karma.conf.js protractor.conf.js tsconfig.json tslint.json $APP_DIR
```

Comment:
There should be an `RUN npm install` here. The completed image should have a fully built version of the web app.